process/factory/shard/vmContainerFactory.go

```diff
@@ -61,16 +68,18 @@ func NewVMContainerFactory(args ArgVMContainerFactory) (*vmContainerFactory, err
 	builtinFunctions := blockChainHookImpl.GetBuiltinFunctionNames()
 
 	return &vmContainerFactory{
-		config:                         args.Config,
-		blockChainHookImpl:             blockChainHookImpl,
-		cryptoHook:                     cryptoHook,
-		blockGasLimit:                  args.BlockGasLimit,
-		gasSchedule:                    args.GasSchedule,
-		builtinFunctions:               builtinFunctions,
-		deployEnableEpoch:              args.DeployEnableEpoch,
-		aheadOfTimeGasUsageEnableEpoch: args.AheadOfTimeGasUsageEnableEpoch,
-		arwenV3EnableEpoch:             args.ArwenV3EnableEpoch,
-		ArwenESDTFunctionsEnableEpoch:  args.ArwenESDTFunctionsEnableEpoch,
+		config:                               args.Config,
+		blockChainHookImpl:                   blockChainHookImpl,
+		cryptoHook:                           cryptoHook,
+		blockGasLimit:                        args.BlockGasLimit,
+		gasSchedule:                          args.GasSchedule,
+		builtinFunctions:                     builtinFunctions,
+		deployEnableEpoch:                    args.DeployEnableEpoch,
+		aheadOfTimeGasUsageEnableEpoch:       args.AheadOfTimeGasUsageEnableEpoch,
+		arwenV3EnableEpoch:                   args.ArwenV3EnableEpoch,
+		ArwenESDTFunctionsEnableEpoch:        args.ArwenESDTFunctionsEnableEpoch,
+		ArwenGasManagementRewriteEnableEpoch: args.ArwenGasManagementRewriteEnableEpoch,
+		containers:                           make([]process.VirtualMachinesContainer, 0),
```

Comment:
you have to pass epochNotifier as argument. And you have to Register the vmContainerFactory to the epochNotifier - otherwise EpochConfirmed will not be called for this component.

Reply:
Done.